src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,248 @@
+    SIMPLE_SETTINGS = ["logfile", "driftfile"].freeze
+    AUTH_COMMANDS = ["trustedkey", "requestkey", "controlkey", "keys"].freeze
+    COMMAND_RECORDS = [
+      "server",
+      "peer",
+      "broadcast",
+      "manycastclient",
+      "multicastclient",
+      "manycastserver"
+    ].freeze
+
+    def self.ntp_attributes(*attrs)
+      attrs.each do |attr|
+        define_method(attr) { get_attribute(attr.to_s) }
+        define_method("#{attr}=") { |record| set_attribute(attr.to_s, record) }
+        define_method("delete_#{attr}") { delete_attribute(attr.to_s) }
+      end
+    end
+
+    ntp_attributes :driftfile, :logfile, :keys, :requestkey, :controlkey
+
+    def initialize(path: DEFAULT_PATH, file_handler: nil)
+      super(PARSER, path, file_handler: file_handler)
+    end
+
+    def servers
+      RecordCollection.new(ServerRecord, data.collection("server"))
+    end
+
+    def peers
+      RecordCollection.new(PeerRecord, data.collection("peer"))
+    end
+
+    def restricts
+      RecordCollection.new(RestrictRecord, data.collection("restrict"))
+    end
+
+    private
+
+    def get_attribute(name)
+      return nil if data[name].nil?
+      AttributeRecord.new_from_augeas(data[name])
+    end
+
+    def set_attribute(name, record)
+      data[name] = record.to_augeas
+    end
+
+    def delete_attribute(name)
+      data.delete(name)
+    end
+
+    # class to manage a collection of some ntp entries, as
+    # server, restrict, peer, etc.
+    #   For example:
+    #     server 0.opensuse.pool.ntp.org iburst
+    #     server 1.opensuse.pool.ntp.org iburst
+    #     server 2.opensuse.pool.ntp.org iburst
+    #     server 3.opensuse.pool.ntp.org iburst
+    class RecordCollection
+      include Enumerable
+
+      def initialize(record_class, augeas_collection)
+        @record_class = record_class
+        @augeas_collection = augeas_collection
+      end
+
+      def each
+        @augeas_collection.each do |augeas_value|
+          yield @record_class.new_from_augeas(augeas_value)
+        end
+      end
+
+      def add(record)
+        @augeas_collection.add(record.to_augeas)
+      end
+
+      def delete(record)
+        @augeas_collection.delete(record.to_augeas)
+      end
+
+      def replace(old_record, new_record)
+        return if !include? old_record
+        matcher = CFA::Matcher.new { |_k, v| v == old_record.to_augeas }
+        placer = CFA::ReplacePlacer.new(matcher)
+        @augeas_collection.add(new_record.to_augeas, placer)
+      end
+
+      def empty?
+        @augeas_collection.empty?
+      end
+
+      def ==(other)
+        other.to_a == to_a
+      end
+    end
+
+    # class to represent a general entry of the file.
+    class Record
+      def initialize(value: nil, tree_data: [])
+        @value = value
+        @tree_data = tree_data
+      end
+
+      def self.new_from_augeas(augeas_value)
+        value, tree_data = parse_augeas(augeas_value)
+        record = new
```

Comment:
why not `new(value: value, tree_data: tree_data)`? I really do not like ignoring private accessor.